rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/notifications/UnityAutoSaveConfigureNotification.kt

```diff
@@ -0,0 +1,128 @@
+
+    companion object {
+        private const val settingName = "do_not_show_unity_auto_save_notification"
+    }
+
+    init {
+        solutionLifecycleHost.isBackendLoaded.whenTrue(componentLifetime) {
+            if (!propertiesComponent.getBoolean(settingName) && unityProjectDiscoverer.isUnityProject) {
+
+                val eventMulticaster = EditorFactory.getInstance().eventMulticaster
+                val documentListener: DocumentListener = object : DocumentListener {
+                    override fun documentChanged(event: DocumentEvent) {
+                        if (unityHost.model.editorState.valueOrDefault(EditorState.Disconnected) != EditorState.ConnectedPlay)
+                            return
+
+                        if (!unityHost.model.scriptCompilationDuringPlay.hasValue)
+                            return
+
+                        if (unityHost.model.scriptCompilationDuringPlay.valueOrThrow != ScriptCompilationDuringPlay.RecompileAndContinuePlaying)
+                            return
+
+                        if (!lifetimeDefinition.isAlive)
+                            return
+
+                        val generalSettings = GeneralSettings.getInstance()
+                        if (generalSettings.isAutoSaveIfInactive || generalSettings.isSaveOnFrameDeactivation) {
+
+                            val editor = event.document.getFirstEditor(project) ?: return
+                            showNotification(lifetimeDefinition.lifetime, editor)
+                        }
+                    }
+                }
+
+                eventMulticaster.addDocumentListener(documentListener)
+                lifetimeDefinition.lifetime.onTermination {
+                    eventMulticaster.removeDocumentListener(documentListener)
+                }
+            }
+        }
+    }
+
+    fun showNotification(lifetime: Lifetime, editor: Editor) {
+        application.assertIsDispatchThread()
+
+
+        if (!lifetime.isAlive) return
+        val project = editor.project ?: return
+
+        val textEditor = TextEditorProvider.getInstance().getTextEditor(editor)
+        if (textEditor.getUserData(KEY) != null) return
+
+        textEditor.putUserData(KEY, Any())
+
+        // Do not show notification, when user leaves play mode and start typing in that moment
+        val alarm = SingleAlarm(Runnable {
+
+            if (unityHost.model.editorState.valueOrDefault(EditorState.Disconnected) != EditorState.ConnectedPlay)
+                return@Runnable
+
+            val panel = EditorNotificationPanel(LightColors.RED)
+            panel.setText("You are modifying a script while Unity Editor is being in Play Mode. This can lead to a loss of the state in your running game.")
+
+            panel.createActionLabel("Configure Unity Editor") {
+                unityHost.model.showPreferences.fire()
+                lifetimeDefinition.terminate()
+            }
+
+            panel.createActionLabel("Don't show again") {
```

Comment:
"Do not show again"
Please avoid using "don't", "ain't" and other reductions in UI